akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -473,6 +473,177 @@ private[akka] final case class Conflate[In, Out](seed: In ⇒ Out, aggregate: (O
   override def restart(): Conflate[In, Out] = copy()
 }
 
+private[akka] sealed abstract class AbstractBatch[In, Out](max: Long, costFn: In ⇒ Long, seed: In ⇒ Out,
+                                                           aggregate: (Out, In) ⇒ Out, val in: Inlet[In],
+                                                           val out: Outlet[Out]) extends GraphStage[FlowShape[In, Out]] {
```

Comment:
What I meant with the AST node is that there should be just one class, with one naming of the ports. This would then be Batch, I think, and it would do all the weighting etc. because there is no unweighted implementation.